packages/grafana-data/src/transformations/transformers/groupBy.ts

```diff
@@ -0,0 +1,199 @@
+import { DataTransformerID } from './ids';
+import { DataFrame, FieldType, Field } from '../../types/dataFrame';
+import { DataTransformerInfo } from '../../types/transformations';
+import { getFieldDisplayName } from '../../field/fieldState';
+import { ArrayVector } from '../../vector/ArrayVector';
+import { guessFieldTypeForField } from '../../dataframe/processDataFrame';
+import { reduceField, ReducerID } from '../fieldReducer';
+
+// export interface GroupByFieldOptions {
+//   aggregations: ReducerID[];
+//   groupBy: boolean;
+// }
+
+export interface GroupByTransformerOptions {
+  calculationsByField: Array<[string | null, ReducerID[]]>;
+  // fields: Record<string, GroupByFieldOptions>;
+  byFields: string[];
+}
+
+export const groupByTransformer: DataTransformerInfo<GroupByTransformerOptions> = {
+  id: DataTransformerID.groupBy,
+  name: 'Group By',
+  description: 'Group the data by a field values then process calculations for each group',
+  defaultOptions: {
+    calculationsByField: [],
+    byFields: [],
+  },
+
+  /**
+   * Return a modified copy of the series.  If the transform is not or should not
+   * be applied, just return the input series
+   */
+  transformer: (options: GroupByTransformerOptions) => {
+    const calculationsByField = options.calculationsByField; //.map((val, index) => ({fieldName: val[0], calculations: val[1]}));
+    const groupByFieldNames = options.byFields;
+
+    return (data: DataFrame[]) => {
+      if (options.byFields.length === 0) {
+        return data;
+      }
+
+      const processed: DataFrame[] = [];
+
+      //
+      // First, group the data in a way we can easily work with
+      //
+
+      for (let frame of data) {
+        // Get all the GroupBy fields and put them in the same array
+        const groupByFields: Field[] = [];
+        for (let fieldName of groupByFieldNames) {
+          for (let field of frame.fields) {
+            if (getFieldDisplayName(field) === fieldName) {
+              groupByFields.push(field);
+            }
+          }
+        }
+
+        if (groupByFields.length === 0) {
+          continue; // No group by field in this frame, ignore the frame
+        }
+
+        /*
+         Group together the data (the rows) that have the same values for the GroupBy fields
+         We do that by creating a Map() in which the key will represent the GroupBy values and the object will contain the rows that have this data
+         
+         Something like this :
+        
+          {
+            ["goupByField1Value1","groupByField2Value1", ...] : {fieldName1:Field, fieldname2:Field, ...},
+            ["goupByField1Value2","groupByField2Value2", ...] : {fieldName1:Field, fieldname2:Field, ...},
+            ["goupByField1Value3","groupByField2Value3", ...] : {fieldName1:Field, fieldname2:Field, ...},
+            ...
+          }
+        */
+
+        const groupedData = {};
```

Comment:
```suggestion
        const groupedData: Record<string, Record<string, MutableField>> = {};
```

The following suggestion should fix the errors with the typings. *Note:* you also need to import the `MutableField` by adding `import { MutableField } from '../../dataframe';`.